### PR TITLE
Font Size

### PR DESCRIPTION
**Pull request by jf-kelly**
Added font size functionality for the editor. A dropdown menu button can be found below the editor next to the keyboard shortcuts button.

### CODE REVIEW

**Review by Jlu18 — CHANGES_REQUESTED** [review]
Please change the merging branch to "dev" and fixed the failed unit test.

File: src/actions/userActions.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import * as types from "../constants/ActionTypes";
+
+const googleLoginRef = "apiv1/googlelogins";
+
+export function toggleDarkMode(){
+    return{ type: types.TOGGLE_DARK_MODE };
+}
```
Comment (kdvalin) [inline src/actions/userActions.js]: Isn't this @bryanmontalvan's Dark mode?

If you're relying on his work for a user profile, maybe we should split those changes out such that you don't need to add his code before it's in a finished state

File: src/components/editor/FontSize.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                            defaultValue = {12}
+                            value = {this.props.settings.fontSize}
+                            onChange = {this.handleFontSizeUpdate}>
+                            <option value={12}>12</option>
+                            <option value={18}>18</option>
+                            <option value={20}>20</option>
+                            <option value={25}>25</option>
+                            <option value={28}>28</option>
+                            <option value={35}>35</option>
+                            <option value={50}>50</option>
```
Comment (kdvalin) [inline src/components/editor/FontSize.js:35]: Are we only aiming to support these font sizes?

There should be a better way to have these auto-populate without having to manually add `<option>`s

File: src/containers/Collection.js [hunk 3]
```diff
@@ -28,6 +24,7 @@ Collection.propTypes = {
 const mapStateToProps = state => ({
     editor: state.editor,
     user: state.user.user,
+    userSettings:state.user.settings,
```
Comment (kdvalin) [inline src/containers/Collection.js:27]: add a space between the colon and `state`.  

Reminding myself to update esdoc rules on this @kdvalin.

File: src/containers/Guided.js [hunk 4]
```diff
@@ -21,6 +21,7 @@ Guided.propTypes = {
 const mapStateToProps = state => ({
     editor: state.editor,
     user: state.user.user,
+    usersettings:state.user.settings,
```
Comment (kdvalin) [inline src/containers/Guided.js:24]: Same as previous comment

File: src/containers/Ide.js [hunk 5]
```diff
@@ -21,6 +21,7 @@ Ide.propTypes = {
 const mapStateToProps = state => ({
     editor: state.editor,
     user: state.user.user,
+    usersettings:state.user.settings,
```
Comment (kdvalin) [inline src/containers/Ide.js:24]: Same as previous comment

File: src/reducers/user.js [hunk 6]
```diff
@@ -2,24 +2,50 @@ import * as types from "../constants/ActionTypes";
...
+                    ...state.settings,
+                    darkMode: !state.settings.darkMode
+                }
+            };
+        case types.UPDATE_FONT_SIZE:
+            return{
+                ...state,
+                settings: {
+                    ...state.settings,
+                    fontSize: action.newFont
```
Comment (kdvalin) [inline src/reducers/user.js:38]: rename the `newFont` attribute to something more indicative of what it is, such as `fontSize`

**Review by kdvalin — CHANGES_REQUESTED** [review]
~~Unit tests also need to be fixed~~.  Missing backend PR for user profiles.

**Review by kdvalin — APPROVED** [review]
Approving pending backend PR for User profile API

**Review by Jlu18 — APPROVED** [review]

**Review by Jlu18 — CHANGES_REQUESTED** [review]
I noticed that user settings were not updated the fontsize is changed. You have to use componentDidUpdate to check if usersettings is updated on redux and send the settings to backed accordingly.

**Review by kdvalin — APPROVED** [review]

**Pull request merged**